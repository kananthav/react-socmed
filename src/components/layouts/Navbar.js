import React, { Component } from 'react';
import { Menu } from 'semantic-ui-react';
import { NavLink } from 'react-router-dom';

class Navbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeItem: 'home',
      modalOpen: false
    };

    this.onAddPost = () => {
      this.setState({ modalOpen: true });
    };

    this.onCancelPost = () => {
      this.setState({ modalOpen: false });
    };

    this.onCommitAddPost = values => {
      return this.props.Post.create(values).then(res => {
        this.onCancelPost();
        alert('post created');
      });
    };
  }

  render() {
    const { activeItem } = this.state;

    return (
      <Menu size="large">
        <NavLink to={'/'}>
          <Menu.Item
            color="blue"
            name="home"
            active={activeItem === 'home'}
          />
        </NavLink>
      </Menu>
    )
  }
}

export default Navbar;
