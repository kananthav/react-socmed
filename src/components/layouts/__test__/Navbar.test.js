import React from 'react';
import { shallow } from 'enzyme';

import Navbar from '../Navbar';

describe('<Navbar />', () => {
  it('renders Navbar without crashing', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });
});
