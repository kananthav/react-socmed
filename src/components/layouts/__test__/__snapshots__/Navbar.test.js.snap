// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders Navbar without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <NavLink
        activeClassName="active"
        aria-current="page"
        to="/"
      >
        <MenuItem
          active={true}
          color="blue"
          name="home"
        />
      </NavLink>,
      "size": "large",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": <MenuItem
          active={true}
          color="blue"
          name="home"
        />,
        "to": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "color": "blue",
          "name": "home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          <MenuItem
            active={true}
            color="blue"
            name="home"
          />
        </NavLink>,
        "size": "large",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <MenuItem
            active={true}
            color="blue"
            name="home"
          />,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "color": "blue",
            "name": "home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
